azure-mgmt-network/src/main/java/com/microsoft/azure/management/network/implementation/PointToSiteConfigurationImpl.java

```diff
@@ -0,0 +1,107 @@
+import com.microsoft.azure.management.network.AddressSpace;
+import com.microsoft.azure.management.network.PointToSiteConfiguration;
+import com.microsoft.azure.management.network.VirtualNetworkGateway;
+import com.microsoft.azure.management.network.VpnClientConfiguration;
+import com.microsoft.azure.management.network.VpnClientProtocol;
+import com.microsoft.azure.management.resources.fluentcore.arm.models.implementation.ChildResourceImpl;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Implementation for PointToSiteConfiguration and its create and update interfaces.
+ */
+@LangDefinition
+class PointToSiteConfigurationImpl
+        extends ChildResourceImpl<VpnClientConfiguration, VirtualNetworkGatewayImpl, VirtualNetworkGateway>
+        implements
+        PointToSiteConfiguration,
+        PointToSiteConfiguration.Definition<VirtualNetworkGateway.Update>,
+        PointToSiteConfiguration.Update {
+
+    PointToSiteConfigurationImpl(VpnClientConfiguration inner, VirtualNetworkGatewayImpl parent) {
+        super(inner, parent);
+    }
+
+    @Override
+    public String name() {
+        return null;
+    }
+
+    @Override
+    public VirtualNetworkGatewayImpl attach() {
+        this.parent().attachPointToSiteConfiguration(this);
+        return parent();
+    }
+
+    @Override
+    public PointToSiteConfigurationImpl withAddressPool(String addressPool) {
+        List<String> addressPrefixes = new ArrayList<>();
+        addressPrefixes.add(addressPool);
+        inner().withVpnClientAddressPool(new AddressSpace().withAddressPrefixes(addressPrefixes));
+        return this;
+    }
+
+    @Override
+    public PointToSiteConfigurationImpl withRootCertificate(String name, String certificateData) {
+        if (inner().vpnClientRootCertificates() == null) {
+            inner().withVpnClientRootCertificates(new ArrayList<VpnClientRootCertificateInner>());
+        }
+        inner().vpnClientRootCertificates().add(new VpnClientRootCertificateInner().withName(name).withPublicCertData(certificateData));
+        return this;
+    }
+
+    @Override
+    public Update withoutRootCertificate(String name) {
+        if (inner().vpnClientRootCertificates() != null) {
+            for (VpnClientRootCertificateInner certificateInner : inner().vpnClientRootCertificates()) {
+                if (name.equals(certificateInner.name())) {
+                    inner().vpnClientRootCertificates().remove(certificateInner);
+                    break;
+                }
+            }
+        }
+        return this;
+    }
+
+    @Override
+    public PointToSiteConfigurationImpl withAzureCertificate() {
+        return this;
+    }
+
+    @Override
+    public PointToSiteConfigurationImpl withRadiusAuthentication(String serverIPAddress, String serverSecret) {
+        inner().withRadiusServerAddress(serverIPAddress).withRadiusServerSecret(serverSecret);
```

Comment:
Same question from the interface def; does the createOrUpdate payload accepts both Azure connection certification and RADIUS connections? If not, you will need to "reset" the "inner" as you set one or the other.

Reply:
Fixed.